### PR TITLE
Clean up some Deprecation warnings

### PR DESCRIPTION
**Pull request by jsoref**
### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by jvz — APPROVED** [review]
Changes look fine to me.

File: core/src/main/java/jenkins/security/MasterToSlaveCallable.java
Comment (StefanSpieker) [inline core/src/main/java/jenkins/security/MasterToSlaveCallable.java]: Since ch is null, adding it to the Exception does not make any sense.

Comment (jsoref) [reply core/src/main/java/jenkins/security/MasterToSlaveCallable.java]: This is to address the fact that the methods w/ fewer arguments are all deprecated. https://github.com/jenkinsci/remoting/blob/445c5a94c13ea42e6144e5a5db544c1cc23b0b5c/src/main/java/hudson/remoting/ChannelClosedException.java#L28

The alternative is me writing `(cast)null`, which seemed to be even less sensical.

**Comment by alecharp** [comment]
Closing to re-trigger the build.

**Comment by jsoref** [comment]
I'm going to temporarily push random commits to try to figure out why the build consistently fails. When I'm done, I'll put things back. I'm not going to change the commit hashes.

File: core/src/test/java/hudson/ChannelRule.java
Comment (jsoref) [inline core/src/test/java/hudson/ChannelRule.java]: I'm not sure why. But, I'm guessing that this commit is going to fail. Assuming it does, and the one before it succeeds, I'm going to try dropping it and continuing along w/ the remainder.

Comment (jsoref) [reply core/src/test/java/hudson/ChannelRule.java]: For reference, this log shows which commits passed/failed: https://github.com/jenkinsci/jenkins/commits/4d894a0bf3929b1fc6712333a7babcdb6a1a995c and 12ae67d295daa59e507dedc9eb1f2e0fd2189a94 indeed failed.

Comment (jsoref) [reply core/src/test/java/hudson/ChannelRule.java]: It appears to be the lack of `.withMode(Channel.Mode.BINARY)`:

https://github.com/jenkinsci/remoting/blob/01a821470b1657b065beebe421f1dbdd6eb6c0ff/src/main/java/hudson/remoting/Channel.java#L407

**Comment by jsoref** [comment]
@-jvz / @-StefanSpieker, for reference, this is the only change between what was originally reviewed and what's currently building:
https://github.com/jenkinsci/jenkins/compare/4d894a0bf3929b1fc6712333a7babcdb6a1a995c..jsoref:deprecation
It's effectively a backout/drop of the single commit 12ae67d295daa59e507dedc9eb1f2e0fd2189a94

As that commit is problematic, I intend to provide a correct version of it as a standalone PR.

**Comment by jsoref** [comment]
Ok. CI is now happy with this PR. I'm done force-pushing over it.

@jvz / @StefanSpieker, for reference, this is the only change between what was originally reviewed and what's currently building:
https://github.com/jenkinsci/jenkins/compare/4d894a0bf3929b1fc6712333a7babcdb6a1a995c..jsoref:deprecation
It's effectively a backout/drop of the single commit 12ae67d295daa59e507dedc9eb1f2e0fd2189a94

As that commit is problematic, I intend to provide a correct version of it as a standalone PR.

**Review by jvz — APPROVED** [review]

File: core/src/test/java/hudson/UtilTest.java
Comment (StefanSpieker) [inline core/src/test/java/hudson/UtilTest.java]: Why do you not use UTF_8?

File: core/src/test/java/jenkins/model/lazy/FakeMapBuilder.java
Comment (StefanSpieker) [inline core/src/test/java/jenkins/model/lazy/FakeMapBuilder.java]: Again why not UTF_8?

**Review by StefanSpieker — COMMENTED** [review]
I'm not sure about all the SuppressWarnings("deprecation") outside the test classes. I think this might hide things to refactor.

File: core/src/test/java/hudson/UtilTest.java
Comment (jsoref) [reply core/src/test/java/hudson/UtilTest.java]: Both are used in the codebase. If someone wants to make a PR that converts `US_ASCII` to `UTF_8` and bans the former, I'm fine w/ that.

It was admittedly arbitrary, but since the content is just `a`s, it didn't seem worth asking for `UTF-8`.

If people insist, I can add a commit to change this. But, tbh, I'd rather do a PR across the codebase than an extra commit here followed by an extra PR to clean the rest of the codebase.

(If reviewers here are ok w/ me writing the commit, I'd be slightly more inclined to fix these two files here.)

File: core/src/main/java/hudson/model/Queue.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Queue.java]: Does not look like a good cause.
The original code would use the `LegacyCodeCause` while here we replace it with User Cause which is not true. 

TBH I think it is time to just remove the "old queue format" support. This format was already replaced in 2008 when I started using Hudson, and there is no sense to keep this code at all. First relevant diff in Git, did not dive into the SVN history https://github.com/jenkinsci/jenkins/commit/8bb5e7f5b45b7f2994137f74ac87cb3c20388cbb. Any concerns @jenkinsci/core?

File: core/src/test/java/jenkins/model/lazy/FakeMapBuilder.java
Comment (jglick) [reply core/src/test/java/jenkins/model/lazy/FakeMapBuilder.java]: Because the content is in fact guaranteed to be ASCII.

**Review by oleg-nenashev — DISMISSED** [review]
Requesting changes to indicate that I would like to finish the Queue discussion before merge. I plan to submit another pull request which just removes this legacy logic, so maybe it makes sense to just revert that line for now

File: core/src/main/java/hudson/model/Queue.java
Comment (res0nance) [reply core/src/main/java/hudson/model/Queue.java]: We probably should drop it. Perhaps we revert this change and make a follow up change with its removal.

**Comment by alecharp** [comment]
@oleg-nenashev @jsoref what is the status of this discussion? Should we keep this PR or not?

**Comment by jsoref** [comment]
@oleg-nenashev: dropped, can we move forward?

File: core/src/main/java/hudson/model/Computer.java
Comment (jglick) [inline core/src/main/java/hudson/model/Computer.java:324]: Matches `getLogText()` so OK for now, though ultimately I think this is wrong: logs should be in UTF-8, which https://github.com/jenkinsci/jenkins/blob/7edfc7170c56780ccc715ed67cbe58a37d65bafb/core/src/main/java/hudson/slaves/SlaveComputer.java#L156 ought to enforce.

File: core/src/main/java/hudson/model/LargeText.java
Comment (jglick) [inline core/src/main/java/hudson/model/LargeText.java:84]: Why are you bothering? This whole class is `@Deprecated` so javac should not be emitting any deprecation warnings from it to begin with.

File: core/src/main/java/hudson/model/Queue.java
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java:1859]: No you are missing the point of the original! Just delete this whole paragraph as it is now obsolete.

File: core/src/test/java/hudson/logging/LogRecorderTest.java
Comment (jglick) [inline core/src/test/java/hudson/logging/LogRecorderTest.java:118]: Would rather you _not_ suppress the warning here. Either fix the usage (should now be `matches`) or leave it alone.

File: core/src/test/java/hudson/model/DisplayNameListenerTest.java
Comment (jglick) [inline core/src/test/java/hudson/model/DisplayNameListenerTest.java:32]: Again, no, this is just hiding technical debt.

> Does not behave very consistently. Either write a real functional test with {@code JenkinsRule}, or use PowerMock/Mockito.

File: core/src/test/java/hudson/model/JobTest.java
Comment (jglick) [inline core/src/test/java/hudson/model/JobTest.java:25]: ditto

File: core/src/test/java/hudson/model/RunTest.java
Comment (jglick) [inline core/src/test/java/hudson/model/RunTest.java:55]: ditto

File: core/src/test/java/hudson/search/SearchTest.java
Comment (jglick) [inline core/src/test/java/hudson/search/SearchTest.java:42]: Again, better to use the nondeprecated overload rather than suppressing this.

File: core/src/test/java/hudson/tasks/BuildStepCompatibilityLayerTest.java
Comment (jglick) [inline core/src/test/java/hudson/tasks/BuildStepCompatibilityLayerTest.java:19]: This is an example of an appropriate suppression: we are deliberately testing the deprecated variant. Might be worth including a comment on the annotation line reminding readers of that.

File: core/src/test/java/hudson/util/LineEndNormalizingWriterTest.java
Comment (jglick) [inline core/src/test/java/hudson/util/LineEndNormalizingWriterTest.java:40]: We should just delete this copy of the test, since it was moved to Stapler sources.

File: test/src/test/java/hudson/cli/CLIActionTest.java
Comment (jglick) [inline test/src/test/java/hudson/cli/CLIActionTest.java:57]: No! Leave alone, or actually fix as per

> Authentication modes are better defined in code using `JenkinsRule.createDummySecurityRealm` and `MockAuthorizationStrategy`

**Review by jglick — CHANGES_REQUESTED** [review]

**Comment by alecharp** [comment]
@jsoref do you know when you will have time to come back on this PR and address Jesse's change requests? Thanks.

**Comment by alecharp** [comment]
I'm proposing this PR for close as we have no answer from you @jsoref since October and there is a merge conflict now.

File: core/src/main/java/hudson/model/LargeText.java
Comment (jsoref) [reply core/src/main/java/hudson/model/LargeText.java:84]: I'm pretty sure I only added suppressions because of warnings.

File: core/src/test/java/hudson/logging/LogRecorderTest.java
Comment (jsoref) [reply core/src/test/java/hudson/logging/LogRecorderTest.java:118]: I think this test is specifically testing the deprecated function.

**Review by jvz — APPROVED** [review]

File: core/src/main/java/hudson/model/LargeText.java
Comment (jglick) [reply core/src/main/java/hudson/model/LargeText.java:84]: Recheck, should be unnecessary.

File: core/src/main/java/hudson/model/Computer.java
Comment (jglick) [reply core/src/main/java/hudson/model/Computer.java:329]:
```suggestion
        return Util.loadFile(getLogFile(), /* TODO switch agent logs to UTF-8 */ Charset.defaultCharset());
```

File: test/src/test/java/hudson/cli/CLIActionTest.java
Comment (jglick) [inline test/src/test/java/hudson/cli/CLIActionTest.java:54]: What is this accomplishing? It is still deprecated.

**Review by jglick — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by daniel-beck** [comment]
Are there any plans to address (in response or code) the remaining open comments?

**Comment by alecharp** [comment]
@jsoref could you address the comments from Jesse? Thanks.

File: test/src/test/java/hudson/cli/CLIActionTest.java
Comment (jsoref) [reply test/src/test/java/hudson/cli/CLIActionTest.java:54]: Instead of getting a bunch of warnings for importing the classes, one gets a single warning here.

Having 3 lines of warnings for a single offensive line is not better than having one warning for the same line.

**Comment by timja** [comment]
I can't see any blocking comments and Jesse has approved this

**Review by timja — APPROVED** [review]
Looks like an improvement, thanks!

Suppressions aren't ideal but they can fixed in a followup

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
